pkg/cmd/recoveryapiserver/create.go

```diff
@@ -0,0 +1,115 @@
+package recoveryapiserver
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apiserver/pkg/server"
+	"k8s.io/klog"
+
+	"k8s.io/client-go/tools/watch"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/recovery"
+)
+
+type CreateOptions struct {
+	Options
+
+	StaticPodResourcesDir string
+}
+
+func NewCreateCommand() *cobra.Command {
+	o := &CreateOptions{
+		Options:               NewDefaultOptions(),
+		StaticPodResourcesDir: "/etc/kubernetes/static-pod-resources",
+	}
+
+	cmd := &cobra.Command{
+		Use: "create",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Complete()
+			if err != nil {
+				return err
+			}
+
+			err = o.Validate()
+			if err != nil {
+				return err
+			}
+
+			err = o.Run()
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	return cmd
+}
+
+func (o *CreateOptions) Complete() error {
+	err := o.Options.Complete()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *CreateOptions) Validate() error {
+	err := o.Options.Validate()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *CreateOptions) Run() error {
+	ctx, cancel := watch.ContextWithOptionalTimeout(context.TODO(), 0 /*infinity*/)
+	defer cancel()
+
+	signalHandler := server.SetupSignalHandler()
+	go func() {
+		<-signalHandler
+		cancel()
+	}()
+
+	recoveryApiserver := &recovery.Apiserver{
+		PodManifestDir:        o.PodManifestDir,
+		StaticPodResourcesDir: o.StaticPodResourcesDir,
+	}
+
+	err := recoveryApiserver.Create()
+	if err != nil {
+		return fmt.Errorf("failed to create recovery apiserver: %v", err)
+	}
+	// destroy the server when we're done
+	defer func() {
+		if err := recoveryApiserver.Destroy(); err != nil {
+			klog.Errorf("failed to destroy the recovery apiserver")
+		}
+	}()
+
+	kubeconfigPath := path.Join(recoveryApiserver.GetRecoveryResourcesDir(), recovery.AdminKubeconfigFileName)
+	klog.Infof("Waiting for recovery apiserver to be healthy.")
+	klog.Infof("    export KUBECONFIG=%s", kubeconfigPath)
+	klog.Infof("to access the server.")
+
+	err = recoveryApiserver.WaitForHealthz(ctx)
```

Comment:
this will wait infinitely?

Reply:
> this will wait infinitely?

or until health is ok.  That's why we print the `export KUBECONFIG` before we start this.